aom/src/main/java/com/nedap/archie/aom/utils/AOMUtils.java

```diff
@@ -364,4 +368,46 @@ public static boolean isValidADL14Code(String code) {
         }
         return adl14CodePattern.matcher(code).matches();
     }
+
+    /**
+     * Returns the expanded set of members of a value set, replacing any ac-codes with at-codes if this is a flat archetype
+     * replaces each ac-code in valueSet with its expanded list of at-codes in allValueSets.
+     *
+     * @param allValueSets all value sets in the given archetype or component terminology that applies here
+     * @param valueSet the ValueSet to expand
+     * @return the expanded value set, where each ac code has been replaced with a number of at codes.
+     */
+    public static Set<String> getExpandedValueSetMembers(Map<String, ValueSet> allValueSets, ValueSet valueSet) {
+        Set<String> result = new LinkedHashSet<>();
+        for(String member:valueSet.getMembers()) {
+            if(AOMUtils.isValueSetCode(member)) {
+                ValueSet includedValueSet = allValueSets.get(member);
+                if(includedValueSet == null) {
+                    result.add(member);//cannot expand that which we cannot find
+                } else {
+                    //TODO: potential loop here, resulting in StackOverflowexception. Add validation somewhere?
+                    result.addAll(getExpandedValueSetMembers(allValueSets, includedValueSet));
+                }
```

Comment:
Can value sets contain other value sets? I didn't know that. Is that somewhere in the specifications?

Reply:
That is new, and it should go in the standard soon. However this will be rather annoying to use and maintain without the feature for a value set to contain other value sets. See https://openehr.atlassian.net/browse/SPECAM-68 for the discussion.